File: frontend/src/api/tasks/delete.ts
import { apiPost } from "@/api/common/post";
import type { ResponseData, Error } from "@/types/global";
import type { OneTask } from "@/types/api/tasks/types";

export async function apiDeleteTask(task_id: string): Promise<ResponseData> {
  const urlPath = `/tasks/delete/${task_id}`;
  const response = await apiPost(urlPath, {}, "DELETE", 0, {
    Authorization: true,
    UserData: true,
  });

  if (
    !response ||
    response.status !== "SUCCESS" ||
    response.status_code >= 400
  ) {
    console.error("API response does not return delete task!");
    return {
      isValid: false,
      data: response.data as Error,
      additional: response.additional,
    };
  }

  return {
    isValid: true,
    data: response.data as OneTask,
    additional: response.additional,
  };
}

File: backend/api/tasks/delete.py
from typing import cast

from fastapi import APIRouter, Request, Depends, Query
from api.routers import DELETE_TASK
from core.data.response import ResponseApiData, Error
from core.helper.headers import check_required_headers
from core.helper.validators import is_valid_uuid
from core.middleware.basic_authorization import JWTBasicAuthenticationMiddleware
from core.handler.tasks.delete import handler_delete_task
from core.data.user import UserData

router = APIRouter(tags=["Tasks"])


@router.delete(DELETE_TASK, dependencies=[Depends(JWTBasicAuthenticationMiddleware())])
def delete_task(request: Request, task_id: str):
    required_headers = ["UserData"]
    data_header = check_required_headers(request, required_headers)
    if not data_header['is_valid']:
        return ResponseApiData(
            status="ERROR",
            data=data_header['data'],
            status_code=data_header['status_code'],
            additional=None
        ).to_response()

    user_data = cast(UserData, data_header['data'][0]['data'])

    if not is_valid_uuid(task_id):
        return ResponseApiData(
            status="ERROR",
            data={"message": "Task_id nie jest poprawnego formatu uuid."},
            status_code=400,
            additional=None
        ).to_response()

    response = handler_delete_task(user_data, task_id)
    if not response['is_valid']:
        return ResponseApiData(
            status=response['status'],
            data=response['data'],
            status_code=response['status_code'],
            additional=response['additional']
        ).to_response()

    return ResponseApiData(
        status=response['status'],
        data=response['data'],
        status_code=response['status_code'],
        additional=response['additional']
    ).to_response()
